aten/src/ATen/detail/MPSHooksInterface.h

```diff
@@ -28,13 +28,21 @@ struct TORCH_API MPSHooksInterface {
     return false;
   }
 
+  virtual bool isOnMacOS13orNewer() const {
+    return false;
```

Comment:
It's meant to be used with MPSHooks only. If MPS isn't available, running this should return false always:
`at::detail::getMPSHooks().isOnMacOS13orNewer()`.
It's a way to know if MPS supports the APIs that were added from MacOS 13.

Reply:
Sure but why should you always assume you're on macos older than MacOS 13 if MPS is not available? The two sound quite unrelated. So why no always return true?

My point is that given that this is an arbitrary choice, why do we pick one instead of raising an error like we do for the other methods below.

Reply:
I think we should follow the pattern where all functions other than `hasMPS()` should raise an assert, as they should never be called if MPS is not available.
```suggestion
    AT_ERROR("MPS backend is not enabled");
```

Reply:
Updated the PR.